src/plugins/headingAnchor.ts

```diff
@@ -130,6 +130,10 @@ function decorationForHeading(heading: Heading) {
  * @return {HTMLElement}
  */
 function anchorForHeading(heading: Heading) {
+	const existing = document.getElementById(heading.id)
```

Comment:
Interesting. Does this mean that we kept adding anchor elements for existing headings and they piled up? 🤔

Reply:
This is actually still relevant for the anchor that are left behind when a heading is changed / removed.
One can define a [destroy function](https://prosemirror.net/docs/ref/#view.Decoration^widget^spec.destroy) for the widget. This might be relevant as it's still referenced by the intersection observer. Which might interfere with garbage collection.